src/error.rs

```diff
@@ -0,0 +1,57 @@
+use thiserror::Error;
+
+use crate::stream::Stream;
+
+#[derive(Debug, Error)]
```

Comment:
Add partialEq trait

Reply:
io::Error and SendError don't support PartialEq